Lib/http/cookiejar.py

```diff
@@ -532,15 +532,39 @@ def parse_ns_headers(ns_headers):
     return result
 
 
+# only kept for backwards compatibilty.
 IPV4_RE = re.compile(r"\.\d+$", re.ASCII)
+
+def is_ip_like_hostname(text):
+    """Return True if text is a valid hostname in the form of IP address.
+    
+    If text is a valid IPv4 address, retrun True;
+
+    If text is a valid IPv6 address wrapped in [], return True;
+
+    Else, return False.
+
+    """
+    from ipaddress import IPv4Address, IPv6Address
+    # check for IPv4 address
+    try:
+        IPv4Address(text)
+    except ValueError:
+        # check for IPv6 address in []
+        if text.startswith('[') and text.endswith(']'):
+            try:
+                IPv6Address(text[1:-1])
+            except ValueError:
+                return False
+        else:
+            return False # not a IPv6 address in []
+    return True
```

Comment:
After staring more at this code, I suggest having two small helpers, `_is_ipv4_hostname()` and `is_ipv6_hostname()` so that we can also reuse `_is_ipv6_hostname()` later. Something like that:

```py
def _is_ipv4_hostname(text):
    try:
        IPv4Address(text)
    except ValueError:
        return False
    return True

def _is_ipv6_hostname(text):
    if text.startswith('[') and text.endswith(']'):
        try:
            IPv4Address(text[1:-1])
        except ValueError:
            return False
        return True
    return False
```

Ideally, we should have some `ipaddress.is_ipv4(text)` to reduce the indentation but I don't know if it wouldbe used elsewhere.